### PR TITLE
WIP - Office Worker Sees "Approve Payment" Button in Invoice Panel #161698060

### PR DESCRIPTION
**Pull request by donaldthai**
## Description

**WIP**

This PR adds an `Approve Payment` button in the office invoice panel so that the office user can submit HHG invoices for payment.

ToDo:
- It's probably a good idea to merge in other code relating to the invoice stories before review
- Review with design

## Reviewer Notes

Is there anything you would like reviewers to give additional scrutiny?

## Setup

Add any steps or code to run in this section to help others prepare to run your code:

```sh
echo "Code goes here"
```

## Code Review Verification Steps

* [ ] Code follows the guidelines for [Logging](./docs/backend.md#logging)
* [ ] The requirements listed in
 [Querying the Database Safely](./docs/backend.md#querying-the-database-safely)
 have been satisfied.
* Any new migrations/schema changes:
  * [ ] Follow our guidelines for zero-downtime deploys (see [Zero-Downtime Deploys](./docs/database.md#zero-downtime-migrations))
  * [ ] Have been communicated to #dp3-engineering
* [ ] There are no aXe warnings for UI.
* [ ] This works in IE.
* Any new client dependencies (Google Analytics, hosted libraries, CDNs, etc) have been:
  * [ ] Communicated to @willowbl00
  * [ ] Added to the list of [network dependencies](https://github.com/transcom/mymove#client-network-dependencies)
* [ ] Request review from a member of a different team.
* [ ] Have the Pivotal acceptance criteria been met for this change?

## References

* [Pivotal story](tbd) for this change
* [this article](tbd) explains more about the approach used.

## Screenshots

If this PR makes visible UI changes, an image of the finished UI can help reviewers and casual
observers understand the context of the changes. A before image is optional and
can be included at the submitter's discretion.

Consider using an animated image to show an entire workflow instead of using multiple images. You may want to use GIPHY CAPTURE for this! 📸

_Please frame screenshots to show enough useful context but also highlight the affected regions._

### CODE REVIEW

File: src/shared/Invoice/InvoicePanel.jsx [hunk 1]
```diff
@@ -1,31 +1,62 @@
...
-    this.state = {};
+    this.state = {
+      canApprove: null,
+      shipmentState: null,
+    };
   }
 
+  componentDidUpdate(prevProps) {
+    if (!this.state.shipmentState || prevProps.shipmentState !== this.props.shipmentState) {
+      this.setState({ shipmentState: this.props.shipmentState });
```
Comment (tinyels) [inline src/shared/Invoice/InvoicePanel.jsx]: I'm not sure why you are storing this prop value in state. Can you explain?

File: src/scenes/Office/MoveInfo.jsx [hunk 2]
```diff
@@ -312,6 +315,9 @@ class MoveInfo extends Component {
                     shipment={this.props.shipment}
                     serviceAgents={this.props.serviceAgents}
                     surveyError={this.props.shipmentPatchError && this.props.errorMessage}
+                    canApprovePaymentInvoice={
+                      !hhgDelivered || !hhgApproved || !moveApproved || invoiceSuccess || currentTab !== 'hhg'
```
Comment (tinyels) [inline src/scenes/Office/MoveInfo.jsx]: I think all you need to check here is `hhgDelivered`

File: src/shared/Invoice/InvoicePanel.jsx (hunk 1 above)
Comment (donaldthai) [reply src/shared/Invoice/InvoicePanel.jsx]: I think I was following closely what the tasked said. I'd be happy to remove the that state and use `this.props.shipmentState` by it self.

File: src/shared/Invoice/InvoicePanel.jsx [hunk 3]
```diff
@@ -1,31 +1,46 @@
...
     return (
       <div className="invoice-panel">
-        <BasicPanel title={'Invoicing'}>{invoicingContent}</BasicPanel>
+        <BasicPanel title={'Invoicing'}>
+          {isOfficeSite &&
+            this.props.shipmentState === 'DELIVERED' && (
+              <div className="usa-width-one-whole align-right">
+                <button
+                  className="button button-secondary"
+                  disabled={!this.props.canApprove && !isDevelopment}
```
Comment (tinyels) [inline src/shared/Invoice/InvoicePanel.jsx]: This change is needed for  e2e test to pass
```suggestion
                  disabled={!this.props.canApprove || !isDevelopment}
```

**Pull request merged**